### ISSUE TITLE
Screen resolution issue after deep sleep

### ISSUE THREAD

**Issue by ArchiFleKs** [regolith-linux/regolith-desktop#63]
Hi, I'm using regolith on an X1 carbon 6th. I'm not sure if this is related to regolith or gnome-flashback or even just the xf86 intel driver. 

I have an HDPI display with a native resolution of 2560 * 1440 that I run at 1920*1080 with a scaling of 100% in gnome.

Whenever I exit deep sleep I have some scaling issue and wallpaper is glitched. Before the removal of `feh` in i3 config I could just reload it by restarting i3. But now that gnome wallpaper is used it just stay that way. 

If anyone with the same setup as encountered this issue and find a workaround let me know :) 

Kevin

**Comment by ArchiFleKs** [comment]
Apparently this is a known [bug](https://bugs.launchpad.net/ubuntu/+source/ubuntu-gnome-meta/+bug/1726548)

The solution [here](https://askubuntu.com/questions/1073568/preventing-window-resizing-after-resume) if some people have the same issue.

TL;DR: 

```
gsettings set org.gnome.desktop.interface scaling-factor 1
```
This fixed it on my Lenovo X1 carbon

**Comment by kgilmer** [comment]
I appreciate your taste in laptops @ArchiFleKs !  (I have the same, but 1080p screen)

I had noticed this regression from moving from `feh` as well but have been focusing on other issues (mainly supporting new users and working on Xresource consolidation.)  I'll test this fix on my work machine w/ the 4k monitor and if it fixes the issue will add it to the [regolith init script](https://github.com/regolith-linux/regolith-gnome-flashback/blob/master/i3-gnome-flashback).

If you'd like a Regolith laptop sticker for your x1, send me a mailing address to kgilmer at gmail.com and which of the two you prefer.  Re-opening the issue as I'd like to get this fixed in Regolith itself.

![IMG_20190620_073945](https://user-images.githubusercontent.com/49683/60776636-84a57600-a0e2-11e9-9b61-3ed9a3f227e5.jpg)

**Comment by ArchiFleKs** [comment]
@kgilmer Hi, thanks. will do ;).  Also, to fix the same issue with gdm (My login screen was upscaled also). Just do the same as the `gdm` user:

```
dbus-launch gsettings set org.gnome.desktop.interface scaling-factor 1
```

Running those two command, as my user and as gdm fixed all my resolution issues.

**Comment by kgilmer** [comment]
I tried the commands on my work machine but am still experiencing garbled wallpaper due to the screen changes.  I'll keep looking for a fix and keep this issue open until it's resolved.

**Comment by kgilmer** [comment]
This seems like a workaround: https://askubuntu.com/questions/605847/wallpaper-turns-into-distorted-pixels-after-suspend

**Comment by MoreThanLuck** [comment]
I've also been having a few resolution related issues, mostly with connecting an external display. I'm running Regolith on a 2013 rMBP, which I'll occasionally connect to a 1080p TV via HDMI. On disconnect, the scaling is incorrect. So far, I've only solved it in a reboot.

**Comment by kgilmer** [comment]
@MoreThanLuck next time it happens, can you provide a screenshot?

**Comment by MoreThanLuck** [comment]
Of course. Replicated the issue here. Seems like it's changing the scaling for some reason? The before screenshot of zsh is 2880x1800, my native resolution, and the after of Hackernews is half that, 1440x900.

![Screenshot from 2019-07-11 22-28-26](https://user-images.githubusercontent.com/4261596/61101168-7206a600-a42e-11e9-983d-0c80ab375829.png)
![Screenshot from 2019-07-11 22-46-36](https://user-images.githubusercontent.com/4261596/61101172-74690000-a42e-11e9-93f2-f5ad547e5ec6.png)

**Comment by kgilmer** [comment]
Thanks @MoreThanLuck .  I have not tested much with Gnome scaling.  I'll try and reproduce the issue.

**Comment by leosunmo** [comment]
Has any of this changed for you @MoreThanLuck? There's been a few updates since this was raised.

**Comment by SteadBytes** [comment]
I'm experiencing a similar issue on a 7th gen X1 carbon. It also happens when plugging in an external monitor as well as often completely glitching out and requiring a restart (related issue https://github.com/regolith-linux/regolith-desktop/issues/133) . Let me know if there's anything I can do to help :+1:

**Comment by kgilmer** [comment]
@SteadBytes are you able/interested in upgrading to Ubuntu 19.10 to see if the upgrade resolves your issue?

**Comment by SteadBytes** [comment]
@kgilmer Sure, I was planning to give 19.10 a go this weekend so I can test it out then? 😁

**Comment by kgilmer** [comment]
Yep!  Here's some info you may need: https://github.com/regolith-linux/regolith-desktop/wiki/HowTo:-Upgrade-to-Ubuntu-19.10-(Eoan-Ermine)

**Comment by SteadBytes** [comment]
@kgilmer Upgrading to 19.10 seems (for the moment at least) to have resolved the issue when waking from deep sleep. However, the issue is still present when plugging in an external monitor :sweat_smile:

**Comment by kgilmer** [comment]
@SteadBytes , well at least there is some progress!  Can you describe what you see when plugin an external monitor, or provide a screenshot?  And at this time are you running Compton?

**Comment by leosunmo** [comment]
Since the main issue of deep sleep issues has been resolve (finally! This has been killing me), I will move discussions to the ticket for external screens, https://github.com/regolith-linux/regolith-desktop/issues/133.

If you're still having problems, please comment or reopen!

**Comment by SteadBytes** [comment]
@kgilmer Sorry for being so slow getting back to you. I will reply over on #133
